### PR TITLE
[Backend]Login backend test

### PR DESCRIPTION
**Pull request by okasato**
## Finish writing test for login and register and fix code

 - Write test for login and register and pass all tests
  - ./server/__tests__/authentication.js
  - Add ./server/__tests__/config.js for test

- Fix ./server/db/migrations/20180314111434_add_users_table.js
  - Change the length of password from 10 to 255 because hashed password is too long

- Fix addUser.js and findByAccount.js to make them more concise and remove useless part

### CODE REVIEW

File: server/db/users/findByAccount.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ module.exports = knex => {
   return async params => {
     const account = params;
```
Comment (ChristianSchmidt1981) [inline server/db/users/findByAccount.js]: can you please remove the const account and rename directly params

File: server/__tests__/authentication.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    chai.request(app)
+      .post('/login')
+      .set(
+        'Content-Type', 'application/json'
+      )
+      .send({ 
+        account: 'captainkrystal@gmail.com',
+        password: 'hitoomedia'
+       })
+      .end((err, res) => {
```
Comment (ChristianSchmidt1981) [inline server/__tests__/authentication.js]: please rename err to error
and res to response

File: server/__tests__/authentication.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      .post('/register')
+      .set(
+        'Content-Type', 'application/json'
+      )
+      .send({ 
+        name: 'Captain Krystal',
+        email: 'captainkrystal@gmail.com',
+        password: 'hitoomedia'
+       })
+      .end((err, res) => {
```
Comment (ChristianSchmidt1981) [inline server/__tests__/authentication.js]: please rename err to error
and res to response

File: server/__tests__/authentication.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+describe('GET /register', () => {
+  let status, response;
+
+  before(done => {
+    chai.request(app)
+      .get('/register')
+      .set(
+        'Content-Type', 'application/json'
+      )
+      .end((err, res) => {
```
Comment (ChristianSchmidt1981) [inline server/__tests__/authentication.js]: please rename err to error
and res to response

**Review by ChristianSchmidt1981 — APPROVED** [review]
please change the marked code-parts.

File: server/db/users/findById.js [hunk 5]
```diff
@@ -1,8 +1,7 @@
 module.exports = knex => {
   return async params => {
```
Comment (ChristianSchmidt1981) [inline server/db/users/findById.js]: please rename params to id

File: server/db/users/findByAccount.js [hunk 6]
```diff
@@ -1,6 +1,5 @@
 module.exports = knex => {
   return async params => {
```
Comment (ChristianSchmidt1981) [inline server/db/users/findByAccount.js]: please rename params to account

File: server/db/users/addUser.js [hunk 7]
```diff
@@ -1,12 +1,10 @@
 module.exports = knex => {
   return async params => {
```
Comment (ChristianSchmidt1981) [inline server/db/users/addUser.js]: please rename params to user

**Review by ChristianSchmidt1981 — APPROVED** [review]
please rename 3 variables

File: server/db/users/findById.js [hunk 8]
```diff
@@ -1,8 +1,7 @@
...
-    const id = params;
     try {
-      return await knex('users').where('id', id);
+      return await knex('users').where('id', params);
```
Comment (ryutaudo) [inline server/db/users/findById.js]: You should change parmas to id

File: server/tests/authentication.js [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+      .post('/register')
+      .set(
+        'Content-Type', 'application/json'
+      )
+      .send({ 
+        name: 'Captain Krystal',
+        email: 'captainkrystal@gmail.com',
+        password: 'hitoomedia'
+       })
+      .end((error, res) => {
```
Comment (ChristianSchmidt1981) [inline server/tests/authentication.js]: rename res to response

File: server/tests/authentication.js [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+describe('GET /login', () => {
+  let status, response;
+
+  before(done => {
+    chai.request(app)
+      .get('/login')
+      .set(
+        'Content-Type', 'application/json'
+      )
+      .end((error, res) => {
```
Comment (ChristianSchmidt1981) [inline server/tests/authentication.js]: rename res to response

File: server/tests/authentication.js [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+    chai.request(app)
+      .post('/login')
+      .set(
+        'Content-Type', 'application/json'
+      )
+      .send({ 
+        account: 'captainkrystal@gmail.com',
+        password: 'hitoomedia'
+       })
+      .end((error, res) => {
```
Comment (ChristianSchmidt1981) [inline server/tests/authentication.js]: rename res to response

**Review by ChristianSchmidt1981 — APPROVED** [review]
please rename some variables

**Review by ryutaudo — APPROVED** [review]
Looks good.

**Review by ChristianSchmidt1981 — APPROVED** [review]
i like it

**Pull request merged**